network/p2p/tracer/internal/duplicate_msgs_counter_cache_test.go

```diff
@@ -0,0 +1,132 @@
+package internal
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+const defaultDecay = .99
+
+// TestGossipSubDuplicateMessageTrackerCache tests the functionality of the GossipSubDuplicateMessageTrackerCache.
+// It ensures that the Inc and Get methods work as expected, does not test counter decay specifically.
+func TestGossipSubDuplicateMessageTrackerCache(t *testing.T) {
+	cache := duplicateMessageTrackerCacheFixture(t, 100, defaultDecay, zerolog.Nop(), metrics.NewNoopCollector())
+	peerID := unittest.PeerIdFixture(t)
+	// expect count to initially be 0
+	count, err := cache.Get(peerID)
+	require.NoError(t, err)
+	require.Equal(t, 0.0, count)
+
+	// increment the counter for a peer
+	count, err = cache.Inc(peerID)
+	require.NoError(t, err)
+	require.Equal(t, 1.0, count)
+	count, err = cache.Get(peerID)
+	require.NoError(t, err)
+	unittest.RequireNumericallyClose(t, 1.0, count, .1)
+}
+
+// TestGossipSubDuplicateMessageTrackerCache tests the concurrent functionality of the GossipSubDuplicateMessageTrackerCache.
+// It ensures that the Inc and Get methods work as expected, does not test counter decay specifically.
+func TestGossipSubDuplicateMessageTrackerCache_Concurrent_Inc_Get(t *testing.T) {
```

Comment:
Please also add a test for concurrently incrementing the same peer id.

Reply:
https://github.com/onflow/flow-go/pull/5058/commits/07269a38d2d7edc5b906c52f09532269c851560e